openforcefield/utils/toolkits.py

```diff
@@ -2948,6 +2949,161 @@ def compute_partial_charges_am1bcc(self, molecule):
+            the lower atom index is key[0] and the higher is key[1].
+        """
+
+        # Example sqm.out section with WBOs:
+        #  Bond Orders
+        #
+        #   QMMM:    NUM1 ELEM1 NUM2 ELEM2      BOND_ORDER
+        #   QMMM:       2   C      1   C        1.41107532
+        #   QMMM:       3   C      1   C        1.41047804
+        # ...
+        #   QMMM:      15   H     13   H        0.00000954
+        #   QMMM:      15   H     14   H        0.00000813
+        #
+        #            --------- Calculation Completed ----------
+
+        data = open(file_path).read()
+
+        begin_sep = ''' Bond Orders
+ 
+  QMMM:    NUM1 ELEM1 NUM2 ELEM2      BOND_ORDER
+'''
+        end_sep = '''
+
+           --------- Calculation Completed ----------
+'''
+        fbo_lines = data.split(begin_sep)[1].split(end_sep)[0].split('\n')
+        bond_orders = dict()
+        for line in fbo_lines:
+            linesp = line.split()
+            atom_index_1 = int(linesp[1])
+            atom_index_2 = int(linesp[3])
+            bond_order = float(linesp[5])
+            # To make lookup easier, we identify bonds as integer tuples with the lowest atom index
+            # first and the highest second.
+            index_tuple = tuple(sorted([atom_index_1, atom_index_2]))
+            bond_orders[index_tuple] = bond_order
+        return bond_orders
+
+    def assign_fractional_bond_orders(self, molecule, bond_order_model=None):
+        """
+        Update and store list of bond orders this molecule. Bond orders are stored on each
+        bond, in the `bond.fractional_bond_order` attribute.
+
+        .. warning :: This API is experimental and subject to change.
+
+        Parameters
+        ----------
+        molecule : openforcefield.topology.molecule Molecule
+            The molecule to assign wiberg bond orders to
+        bond_order_model : str, optional, default=None
+            The charge model to use. Only allowed value is 'am1-wiberg'. If None, 'am1-wiberg' will be used.
+
+         """
+
+        # Find the path to antechamber
+        # TODO: How should we implement find_executable?
+        ANTECHAMBER_PATH = find_executable("antechamber")
+        if ANTECHAMBER_PATH is None:
+            raise (IOError("Antechamber not found, cannot run "
+                           "AmberToolsToolkitWrapper.assign_fractional_bond_orders()"))
+
+        if len(molecule._conformers) == 0:
+            raise ValueError(
+                "No conformers present in molecule submitted for fractional bond order calculation. Consider "
+                "loading the molecule from a file with geometry already present or running "
+                "molecule.generate_conformers() before calling molecule.assign_fractional_bond_orders"
+            )
+        if len(molecule._conformers) > 1:
+            logger.warning(f"Warning: In AmberToolsToolkitWrapper.assign_fractional_bond_orders: "
```

Comment:
how slow would it be to generate the bond order for all of the supplied conformers and average over them in some way? Could this even be done in parallel?

Reply:
I think conformer averaging would be a nice thing to do. But, like ELF10, it's not clear that there's a single common-sense way to do it. So, I'd like to leave conformer averaging algorithms as something for the community to experiment with. Once someone wants to try it, they can invent a new keyword for `bond_order_model`, and implement support for it in a ToolkitWrapper.

Reply:
Definitely seems like a thing to do separately/unresolved science.